protocol/messenger_share_urls.go

```diff
@@ -177,8 +177,14 @@ func (m *Messenger) parseCommunityURLWithData(data string, chatKey string) (*URL
 		return nil, err
 	}
 
+	var urlDataProto protobuf.URLData
+	err = proto.Unmarshal(communityData, &urlDataProto)
+	if err != nil {
+		return nil, err
+	}
+
 	var communityProto protobuf.Community
-	err = proto.Unmarshal(communityData, &communityProto)
+	err = proto.Unmarshal(urlDataProto.Content, &communityProto)
```

Comment:
Btw, the `protobuf.URLData` wrapper may be redundant. The spec went trough several changes, and if I remember I had it there when I planned for it to be extensible for fields like timestamp for when the url has been encoded or whether it caries content for community, channel or user.

Let me know if you'd prefer dropping it, so we can coordinate the change across the board.

Reply:
Just make sure, I opened this thread on decoding but encoding should be changed accordingly too.

Reply:
Imho, it's better to have only one `unmarshal', but I don't insist on it

Reply:
> Should I wait until this PR is ready, or create another one for the main status-go repo?

@MishkaRogachev this is main status-go repo, isn't?

Also, I left this in the PR's description.

> note: this is just a draft for others who worked on go implemtation to take over, feel free to push commits to it if the settings allow otherwise please let me know

I see you're assigned now, so do as you see fit 🙂, however I think it'd be easiest for you to just push new commits here. Settings should allow it.

<img width="830" alt="image" src="https://github.com/status-im/status-go/assets/13265126/13a4e783-1a61-4bc3-8cf1-c97ded72daca">

I just wanted to help you out finding the issue.

Reply:
I was a little confused because it's a fork but not a branch. Sure, i will continue this PR here, however i need test data for other url types. Should i take it from [here](https://github.com/status-im/status-web/blob/main/packages/status-js/src/utils/create-url.test.ts)?

Reply:
Just noting that https://github.com/status-im/status-web/pull/461/files#diff-378ed45ccbd1b990783b9cc2f2ee73430328530f17339176b4fa658f2b6eda08 hash been merged so the "test data" is our `main` too for reference.

Reply:
Thanks for pointing that out, i'm will update the PR soon